pkg/skaffold/kubernetes/manifest/images.go

```diff
@@ -20,24 +20,27 @@ import (
 	"context"
 	"strconv"
 
+	apimachinery "k8s.io/apimachinery/pkg/runtime/schema"
+
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/graph"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/instrumentation"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/output/log"
+	latestV1 "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest/v1"
 )
 
 // GetImages gathers a map of base image names to the image with its tag
-func (l *ManifestList) GetImages() ([]graph.Artifact, error) {
+func (l *ManifestList) GetImages(allowlist map[apimachinery.GroupKind]latestV1.ResourceFilter, denylist map[apimachinery.GroupKind]latestV1.ResourceFilter) ([]graph.Artifact, error) {
```

Comment:
can we have a type:
```go
type resourceSelector interface {
    allow(apimachinery.GroupKind) bool
}
```
and use:
```go
func (l *ManifestList) GetImages(r resourceSelector) ([]graph.Artifact, error) {
```
here, and everywhere else?

Reply:
Done.  Currently using:
```
type ResourceSelector interface {
	allowByGroupKind(apimachinery.GroupKind) bool
	allowByNavpath(apimachinery.GroupKind, string, string) (string, bool)
}
```